### DISCUSSION TITLE
I'm experiencing Stockfish 16.1 failing to respond after sending it a greatly imbalanced position

### DISCUSSION THREAD

**Discussion by Bookup2** [official-stockfish/Stockfish#5171] [Q&A] [answered]
Here is the log file which includes the problematic FEN. You can see that SF 16.1 fails to reply after the interface sends the FEN.

====4/13/2024 10:51:15 AM====
--> [00126287.562] Stockfish 16.1 by the Stockfish developers (see AUTHORS file)
<-- [00126287.578] uci
--> [00126287.765] id name Stockfish 16.1
--> [00126287.765] id author the Stockfish developers (see AUTHORS file)
--> [00126287.765] 
--> [00126287.765] option name Debug Log File type string default 
--> [00126287.765] option name Threads type spin default 1 min 1 max 1024
--> [00126287.765] option name Hash type spin default 16 min 1 max 33554432
--> [00126287.765] option name Clear Hash type button
--> [00126287.765] option name Ponder type check default false
--> [00126287.765] option name MultiPV type spin default 1 min 1 max 256
--> [00126287.765] option name Skill Level type spin default 20 min 0 max 20
--> [00126287.765] option name Move Overhead type spin default 10 min 0 max 5000
--> [00126287.765] option name nodestime type spin default 0 min 0 max 10000
--> [00126287.765] option name UCI_Chess960 type check default false
--> [00126287.765] option name UCI_LimitStrength type check default false
--> [00126287.765] option name UCI_Elo type spin default 1320 min 1320 max 3190
--> [00126287.765] option name UCI_ShowWDL type check default false
--> [00126287.765] option name SyzygyPath type string default <empty>
--> [00126287.765] option name SyzygyProbeDepth type spin default 1 min 1 max 100
--> [00126287.765] option name Syzygy50MoveRule type check default true
--> [00126287.765] option name SyzygyProbeLimit type spin default 7 min 0 max 7
--> [00126287.765] option name EvalFile type string default nn-b1a57edbea57.nnue
--> [00126287.765] option name EvalFileSmall type string default nn-baff1ede1f90.nnue
--> [00126287.765] uciok
<-- [00126287.796] isready
--> [00126287.968] readyok
<-- [00126288.015] stop
<-- [00126288.015] setoption name MultiPV value 3
<-- [00126288.015] position fen 4k3/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 0
<-- [00126288.015] go infinite
<-- [00126361.734] stop
<-- [00126363.171] setoption name MultiPV value 3
<-- [00126363.828] position fen rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 0
<-- [00126364.531] go infinite
<-- [00126562.828] stop
<-- [00126564.843] quit

**Comment by peregrineshahin** [comment]
Please send the fen correctly without the castling rights for black, because black can't castle
```diff
-  position fen 4k3/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 0
+  position fen 4k3/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQ - 0 1
```

**Reply by Bookup2** [reply]
You may have found the problem! I will adjust that and try again.

Still, should SF stop replying to pipe commands after being sent an invalid FEN?

**Reply by peregrineshahin** [reply, marked as answer]
Yes, it will crash.
see also:
https://github.com/official-stockfish/Stockfish/issues/5131
https://github.com/official-stockfish/Stockfish/pull/4563
https://github.com/official-stockfish/Stockfish/wiki/Stockfish-FAQ#stockfish-crashed

**Reply by Bookup2** [reply]
That explains it. Fixing the FEN castling rights got it working.

I'm debugging Chess Openings Wizard which gets confused when SF reports centipawn evals greater than 1000.
